### PR TITLE
Do not assume that model id is int

### PR DESCRIPTION
**Pull request by teohhanhui**
| Q               | A
| --------------- | -----
| Branch?         | master
| Bug fix?        | yes
| New feature?    | no
| BC breaks?      | yes (scalar type hint)
| Deprecations?   | no
| Related tickets | follow up for #8544
| License         | MIT

### CODE REVIEW

**Comment by teohhanhui** [comment]
There's 1 place where there's BC break because of the return type hint being removed. It means we promised before that it should return an int, but now we revoke that promise. The client code might not have handled it.

**Comment by stefandoorn** [comment]
Isn't `null|int` and `getId(): ?int` better in this case? Sylius models don't use UUID's or something as far as I recall. Or is the idea to make it possible to do something like that? I currently rely on the fact that it's an ID, which won't change of course with this change but it might become more unclear what's being returned (less strict contract).

**Comment by teohhanhui** [comment]
The idea is that Sylius should not care whether `id` is an `int` or not. It should not matter. Changing the contract means we expect that Sylius and extensions should not assume that `id` is `int`.

**Comment by teohhanhui** [comment]
About the BC promise, maybe we're fine because of http://docs.sylius.com/en/latest/contributing/code/bc.html#models-model-interfaces

File: src/Sylius/Bundle/PayumBundle/Model/GatewayConfig.php [hunk 1]
```diff
@@ -19,14 +19,14 @@
...
-     * @var int
+     * @var mixed
      */
     protected $id;
 
     /**
-     * @return int
+     * {@inheritdoc}
      */
-    public function getId(): int
```
Comment (pamil) [inline src/Sylius/Bundle/PayumBundle/Model/GatewayConfig.php]: Need to check whether it is a BC break or not.

**Comment by teohhanhui** [comment]
How do we proceed with this one?

**Comment by lchrusciel** [comment]
Hello Han, 

I know it is been a while since we moved anything in this PR. IMHO removing `int` typehint is not a BC break and we could have it finally merged. Could you rebase to the newest master?

**Pull request closed without merging**